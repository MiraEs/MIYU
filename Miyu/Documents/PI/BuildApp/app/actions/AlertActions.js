'use strict';
import { createAction } from 'redux-actions';
import EventEmitter from '../lib/eventEmitter';

function showAlertAction(message, type, button, callback, bannerVisibleTimeout) {
	EventEmitter.emit('showScreenAlert', {
		message,
		type,
		button,
		callback,
		bannerVisibleTimeout,
	});
}

const showAlert = createAction('SHOW_ALERT_ACTION', showAlertAction);

module.exports = {
	showAlert,
};
