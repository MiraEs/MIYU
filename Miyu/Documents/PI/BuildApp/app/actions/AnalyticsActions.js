'use strict';
import tracking from '../lib/analytics/tracking';
import TrackingActions from '../lib/analytics/TrackingActions';
import { createAction } from 'redux-actions';

function trackStateHelper(state, data) {
	tracking.trackState(state, data);
	return {
		state,
		data,
	};
}

const trackState = createAction('ANALYTICS_TRACK_STATE', trackStateHelper);

function trackActionHelper(action, data) {
	switch (action) {
		case TrackingActions.FAVORITE_ADD_TO_CART:
		case TrackingActions.PDP_ADDITEM:
		case TrackingActions.CART_ITEM_ADD:
		case TrackingActions.RELATED_PRODUCT_REQUIRED_ADD_TO_CART:
		case TrackingActions.RELATED_PRODUCT_RECOMMENDED_ADD_TO_CART:
		case TrackingActions.RELATED_PRODUCT_MAY_WE_SUGGEST_ADD_TO_CART:
			tracking.trackAddToCart(action, data);
			break;
		case TrackingActions.CUSTOMER_SIGNUP_COMPLETE:
			tracking.trackCustomerRegistered(data.user, data.cart, data.methodName);
			break;
		default:
			tracking.trackAction(action, data);
			break;
	}
	return {
		action,
		data,
	};
}

const trackAction = createAction('ANALYTICS_TRACK_ACTION', trackActionHelper);

module.exports = {
	trackStateHelper,
	trackState,
	trackActionHelper,
	trackAction,
};
