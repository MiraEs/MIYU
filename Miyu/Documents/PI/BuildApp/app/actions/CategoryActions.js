import categoryService from '../services/categoryService';

import {
	CATEGORIES_SUCCESS,
	CATEGORY_SUCCESS,
	CATEGORIES_FAIL,
} from '../constants/categoryConstants';
import { createAction } from 'redux-actions';
import helpers from '../lib/helpers';


const categoryActions = {
	getCategoriesSuccess: createAction(CATEGORIES_SUCCESS),
	getCategorySuccess: createAction(CATEGORY_SUCCESS),
	getCategoriesFail: createAction(CATEGORIES_FAIL),

	getCategories() {
		return (dispatch) => {
			return categoryService.getCategories().then((data) => {
				helpers.serviceErrorCheck(data);
				dispatch(categoryActions.getCategoriesSuccess(data));
			})
				.catch((error) => {
					dispatch(categoryActions.getCategoriesFail(error.message));
				});
		};
	},

	getCategory(category) {
		return (dispatch) => {
			return Promise.all([category.name ? category : categoryService.getCategory(category.id), categoryService.getSubCategories(category.id)]).then(([category, subCategories]) => {
				dispatch(categoryActions.getCategorySuccess({ category, subCategories }));
			}, (error) => {
				dispatch(categoryActions.getCategoriesFail(error));
			});
		};
	},

};

module.exports = categoryActions;
