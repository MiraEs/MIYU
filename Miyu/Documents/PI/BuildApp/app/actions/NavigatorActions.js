import { createAction } from 'redux-actions';
import { NavigationActions } from '@expo/ex-navigation';
import store from '../store/configStore';

const navigatorPop = createAction('NAVIGATION_POP', (navUID) => {
	store.dispatch(NavigationActions.pop(navUID));
});

const navigatorPush = createAction('NAVIGATION_PUSH', (route, navUID) => {
	store.dispatch(NavigationActions.push(navUID, route));
});

const navigatorPopToTop = createAction('NAVIGATION_POP_TO_TOP', (navUID) => {
	store.dispatch(NavigationActions.popToTop(navUID));
});

const navigatorReplace = createAction('NAVIGATION_REPLACE', (route, navUID) => {
	store.dispatch(NavigationActions.replace(navUID, route));
});

const navigatorPopToRoute = createAction('NAVIGATOR_POP_TO_ROUTE', (routeName, navUID) => {
	let matchingIndex = 0;
	const currentRoutes = store.getState().navigation.navigators[navUID].routes;
	const matchingRoute = currentRoutes.find((route, index) => {
		if (route.routeName === routeName) {
			matchingIndex = index;
			return true;
		}
		return false;
	});
	if (matchingRoute) {
		store.dispatch(NavigationActions.popN(navUID, (currentRoutes.length - 1) - matchingIndex));
	}
});

module.exports = {
	navigatorPop,
	navigatorPush,
	navigatorReplace,
	navigatorPopToRoute,
	navigatorPopToTop,
};
