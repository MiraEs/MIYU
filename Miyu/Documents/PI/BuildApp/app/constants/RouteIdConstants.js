export const CHECKOUT_SCREEN = 'checkout';
export const CHECKOUT_CONFIRMATION = 'checkoutConfirmation';
export const CREDIT_CARD_SCREEN = 'CREDIT_CARD_SCREEN';
export const ACCOUNT_DETAILS_SCREEN = 'accountDetails';
export const PAYPAL_SCREEN = 'PAYPAL_SCREEN';
export const CART_SCREEN = 'CART_SCREEN';
