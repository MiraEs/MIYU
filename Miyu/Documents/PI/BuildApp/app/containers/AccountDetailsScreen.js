import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import {
	Alert,
	View,
	Switch,
	StyleSheet,
} from 'react-native';
import {
	Text,
	ListView,
	Image,
	Button,
	withScreen,
} from 'BuildLibrary';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import {
	deleteCreditCard,
	getCreditCards,
} from '../actions/CheckoutActions';
import {
	updateCustomer,
	getCustomerAddresses,
	deleteCustomerAddress,
} from '../actions/UserActions';
import styles from '../lib/styles';
import helpers from '../lib/helpers';
import {
	CREATE_ADDRESS,
	CREATE_CREDIT_CARD,
	DELETE_ADDRESS,
	DELETE_CREDIT_CARD,
	EDIT_ADDRESS,
	EDIT_CREDIT_CARD,
} from '../lib/analytics/TrackingActions';
import TappableListItem from '../components/TappableListItem';
import ListHeader from '../components/listHeader';
import Icon from 'react-native-vector-icons/Ionicons';
import {
	SHIPPING_ADDRESS,
	STATE_EDIT_ADDRESS,
	STATE_ADD_ADDRESS,
} from '../constants/Addresses';
import {
	getCardImage,
	parseCardTypeName,
} from '../lib/CreditCard';
import { ACCOUNT_DETAILS_SCREEN } from '../constants/RouteIdConstants';
import Address from '../components/Address';
import { navigatorPopToRoute } from '../actions/NavigatorActions';

import { showAlert } from '../actions/AlertActions';

const componentStyles = StyleSheet.create({
	headerContainer: {
		padding: styles.measurements.gridSpace3,
		alignItems: 'center',
	},
	sectionContainer: {
		padding: styles.measurements.gridSpace1,
	},
	firstRow: {
		marginTop: styles.measurements.gridSpace1,
	},
	newButton: {
		width: styles.dimensions.width / 2,
		flex: 0,
		marginLeft: styles.measurements.gridSpace1,
		marginBottom: styles.measurements.gridSpace1,
	},
	switchRow: {
		flexDirection: 'row',
		alignItems: 'center',
		paddingVertical: styles.measurements.gridSpace2,
	},
	switch: {
		marginTop: styles.measurements.gridSpace2,
	},
	rowButtons: {
		width: 52,
		maxWidth: 52,
		marginLeft: styles.measurements.gridSpace1,
	},
	objectRow: {
		paddingBottom: styles.measurements.gridSpace3,
		paddingHorizontal: styles.measurements.gridSpace1,
	},
	defaultText: {
		paddingTop: styles.measurements.gridSpace1,
	},
	paymentImage: {
		marginRight: styles.measurements.gridSpace1,
		backgroundColor: styles.colors.white,
	},
	cardTypeInfo: {
		flexDirection: 'row',
		alignItems: 'center',
	},
});

export class AccountDetailsScreen extends Component {

	constructor(props) {
		super(props);

		this.defaultAccountDetails = {
			addresses: [{
				component: this.renderNewAddressButton,
			}],
			payments: [{
				component: this.renderNewCreditCardButton,
			}],
			'email subscriptions': [{
				component: this.renderEmailSubscriptions,
			}],
			'account settings': [{
				component: this.renderAccountSettingsRows,
			}],
		};
		this.ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2, sectionHeaderHasChanged: (s1, s2) => s1 !== s2});

		this.state = {
			optimistic: false,
			dataSource: this.ds.cloneWithRowsAndSections(this.defaultAccountDetails),
		};

		this.optimism = {
			isSubscriber: !props.customer.isSubscriber,
		};
	}

	componentWillReceiveProps(newProps) {
		const dataSource = {
			addresses: [
				...this.getSortedAddresses(newProps),
				...this.defaultAccountDetails.addresses,
			],
			payments: [
				...this.getCreditCards(newProps),
				...this.defaultAccountDetails.payments,
			],
			'email subscriptions': this.defaultAccountDetails['email subscriptions'],
			'account settings': this.defaultAccountDetails['account settings'],
		};

		this.setState({ dataSource: this.ds.cloneWithRowsAndSections(dataSource) });
	}

	setScreenTrackingInformation() {
		return {
			name: 'build:app:more:acct',
		};
	}

	getScreenData = () => {
		const { actions } = this.props;
		actions.getCreditCards().catch(helpers.noop).done();
		actions.getCustomerAddresses().catch(helpers.noop).done();
	};

	getCreditCards = (props) => {
		let { creditCards } = props;
		const { defaultCard } = props;

		if (defaultCard && defaultCard.creditCardId) {
			defaultCard.isDefault = true;
			creditCards = [defaultCard].concat(creditCards);
		}

		return creditCards;
	};

	getSortedAddresses = (props) => {
		const { addresses, defaultShippingAddressId } = props;
		return addresses
			.sort((a, b) => b.addressId - a.addressId)
			.sort((a, b) => b.addressId === defaultShippingAddressId ? Number.POSITIVE_INFINITY : 0);
	};

	addOrEditAddress = (address) => {
		const addressScreenState = address ? STATE_EDIT_ADDRESS : STATE_ADD_ADDRESS;
		this.props.navigator.push('newAddress', {
			address,
			addressScreenState,
			title: address ? 'Edit Address' : 'New Address',
			addressTypeId: SHIPPING_ADDRESS,
			onSaveSuccess: this.navigateToAccountDetails,
		});
	};

	changePassword = () => {
		this.props.navigator.push('changePassword');
	};

	deleteCreditCard = (creditCardId) => {
		Alert.alert(
			'Delete Credit Card',
			'Are you sure you want to delete your card?',
			[
				{ text: 'No' },
				{
					text: 'Yes',
					onPress: () => {
						this.props.actions.deleteCreditCard(creditCardId)
							.then(() => {
								this.props.actions.showAlert('Successfully deleted Credit Card');
							})
							.catch(() => {
								this.props.actions.showAlert('Failed to delete Credit Card', 'error');
							})
							.done();
					},
				},
			]
		);
	};

	deleteCustomerAddress = (addressId) => {
		Alert.alert(
			'Delete Address',
			'Are you sure you want to delete your address?',
			[
				{ text: 'No' },
				{
					text: 'Yes',
					onPress: () => {
						this.props.actions.deleteCustomerAddress(addressId)
							.then(() => {
								this.props.actions.showAlert('Successfully deleted address');
							})
							.catch(() => {
								this.props.actions.showAlert('Failed to delete address', 'error');
							})
							.done();
					},
				},
			]
		);
	};

	editDetails = () => {
		this.props.navigator.push('editAccountDetails');
	};

	onPressNewCardButton = () => {
		this.props.navigator.push('creditCardScreen', {
			hideTabs: false,
			onSaveSuccess: this.navigateToAccountDetails,
			isCheckout: false,
		});
	};

	navigateToAccountDetails = () => {
		navigatorPopToRoute(ACCOUNT_DETAILS_SCREEN, 'more');
	};

	toggleEmailSubscription = (isSubscriber) => {
		this.setState({ optimistic: true });
		this.props.actions.updateCustomer({ isSubscriber })
			.then(() => {
				this.setState({ optimistic: false });
				this.optimism.isSubscriber = !this.optimism.isSubscriber;
			})
			.catch(() => {
				this.setState({ optimistic: false });
			})
			.done();
	};

	updateCreditCard = (creditCardId) => {
		this.props.navigator.push('creditCardScreen', {
			title: 'Update Card',
			hideTabs: false,
			onSaveSuccess: this.navigateToAccountDetails,
			isCheckout: false,
			isEdit: true,
			creditCardId,
		});
	};

	renderAccountSettingsRows = () => {
		return (
			<View>
				<TappableListItem
					accessibilityLabel="Change Password"
					body="Change Password"
					onPress={this.changePassword}
				/>
				<TappableListItem
					accessibilityLabel="Edit Account Details"
					body="Edit Account Details"
					onPress={this.editDetails}
				/>
			</View>
		);
	};

	renderAddress = (address, index) => {
		const style = !index ? componentStyles.firstRow : null;

		return (
			<View style={[componentStyles.objectRow, style]}>
				<View style={styles.elements.centeredFlexRow}>
					<Address
						address={address}
						style={styles.elements.flex}
					/>
					<View style={styles.elements.centeredFlexRow}>
						<Button
							style={componentStyles.rowButtons}
							color="white"
							accessibilityLabel="Edit Address"
							onPress={() => this.addOrEditAddress(address)}
							trackAction={EDIT_ADDRESS}
						>
							<Icon
								size={30}
								name="md-create"
								color={styles.colors.secondary}
							/>
						</Button>
						<Button
							style={componentStyles.rowButtons}
							color="white"
							accessibilityLabel="Remove Address"
							onPress={() => this.deleteCustomerAddress(address.addressId)}
							trackAction={DELETE_ADDRESS}
						>
							<Icon
								size={30}
								name="ios-trash"
								color={styles.colors.secondary}
							/>
						</Button>
					</View>
				</View>
			</View>
		);
	};

	renderCreditCard = (creditCard, index) => {
		const style = !index ? componentStyles.firstRow : null;

		return (
			<View style={[componentStyles.objectRow, style]}>
				<View style={styles.elements.centeredFlexRow}>
					<View style={styles.elements.flex}>
						<View style={componentStyles.cardTypeInfo}>
							<Image
								style={componentStyles.paymentImage}
								source={getCardImage(parseCardTypeName(creditCard.cardType))}
								resizeMode="contain"
								height={60}
								width={60}
							/>
							<Text>****{creditCard.lastFour}</Text>
						</View>
						<Text lineHeight={false}>Expires {helpers.getFormattedDate(creditCard.expDate, 'MM/YY')}</Text>
					</View>
					{this.renderCreditCardEditDeleteButtons(creditCard)}
				</View>
				{this.renderIsDefaultCard(creditCard)}
			</View>
		);
	};

	renderCreditCardEditDeleteButtons = ({ creditCardId }) => {
		return (
			<View style={styles.elements.centeredFlexRow}>
				<Button
					style={componentStyles.rowButtons}
					color="white"
					accessibilityLabel="Edit Card"
					onPress={() => this.updateCreditCard(creditCardId)}
					trackAction={EDIT_CREDIT_CARD}
				>
					<Icon
						size={30}
						name="md-create"
						color={styles.colors.secondary}
					/>
				</Button>
				<Button
					style={componentStyles.rowButtons}
					color="white"
					accessibilityLabel="Remove Card"
					onPress={() => this.deleteCreditCard(creditCardId)}
					trackAction={DELETE_CREDIT_CARD}
				>
					<Icon
						size={30}
						name="ios-trash"
						color={styles.colors.secondary}
					/>
				</Button>
			</View>
		);
	};

	renderCustomerInfo = () => {
		const { customer } = this.props;
		return (
			<View style={componentStyles.headerContainer}>
				{this.renderProBadge()}
				<Text weight="bold">{customer.firstName} {customer.lastName}</Text>
				<Text>{customer.email}</Text>
			</View>
		);
	};

	renderEmailSubscriptions = () => {
		const {customer} = this.props;
		return (
			<View style={componentStyles.sectionContainer} >
				<View style={componentStyles.switchRow}>
					<View style={styles.elements.flex}>
						<Text>Promotional Emails</Text>
						<Text size="small">Get notified on the best deals and coupons.</Text>
					</View>
					<Switch
						style={componentStyles.switch}
						onTintColor={styles.colors.primary}
						value={this.state.optimistic ? this.optimism.isSubscriber : customer.isSubscriber}
						onValueChange={this.toggleEmailSubscription}
					/>
				</View>
			</View>
		);
	};

	renderIsDefaultCard = (creditCard) => {
		if (creditCard.isDefault) {
			return (
				<Text
					color="accent"
					style={componentStyles.defaultText}
				>
					PREFERRED CARD
				</Text>
			);
		}
	};

	renderNewAddressButton = () => {
		return (
			<Button
				color="white"
				text="New Address"
				style={componentStyles.newButton}
				accessibilityLabel="New Address"
				textColor="secondary"
				onPress={this.addOrEditAddress}
				trackAction={CREATE_ADDRESS}
			/>
		);
	};

	renderNewCreditCardButton = () => {
		return (
			<Button
				color="white"
				textColor="secondary"
				text="New Card"
				style={componentStyles.newButton}
				accessibilityLabel="New Card"
				onPress={this.onPressNewCardButton}
				trackAction={CREATE_CREDIT_CARD}
			/>
		);
	};

	renderProBadge = () => {
		const { customer } = this.props;
		if (customer.isPro) {
			return (
				<Image
					source={require('../images/pro_stamp.png')}
					resizeMode="contain"
					size="none"
				/>
			);
		}
	};

	renderRows = (rowData, sectionId, rowId) => {
		if (typeof rowData.component === 'function') {
			return rowData.component();
		}

		if (sectionId === 'addresses') {
			return this.renderAddress(rowData, parseInt(rowId, 10));
		} else if (sectionId === 'payments') {
			return this.renderCreditCard(rowData, parseInt(rowId, 10));
		}
	};

	renderSectionHeader = (sectionData, sectionID) => {
		return <ListHeader text={sectionID.toUpperCase()} />;
	};

	render() {
		return (
			<View
				style={styles.elements.screen}
			>
				<ListView
					dataSource={this.state.dataSource}
					enableEmptySections={false}
					renderHeader={this.renderCustomerInfo}
					renderRow={this.renderRows}
					renderSectionHeader={this.renderSectionHeader}
					scrollsToTop={true}
				/>
			</View>
		);
	}
}

AccountDetailsScreen.route = {
	navigationBar: {
		title: 'Account Details',
	},
};

AccountDetailsScreen.propTypes = {
	customer: PropTypes.object.isRequired,
	creditCards: PropTypes.array,
	defaultCard: PropTypes.object,
	addresses: PropTypes.array,
	loading: PropTypes.bool,
	actions: PropTypes.object,
	defaultShippingAddressId: PropTypes.number,
	navigator: PropTypes.shape({
		push: PropTypes.func,
	}),
};

AccountDetailsScreen.defaultProps = {
	customer: {},
	creditCards: [],
	addresses: [],
};

const mapStateToProps = (state) => {
	return {
		customer: state.userReducer.user,
		creditCards: state.checkoutReducer.creditCards,
		defaultCard: state.checkoutReducer.defaultCard,
		addresses: state.userReducer.user.shippingAddresses,
		loading: state.userReducer.isLoading,
		defaultShippingAddressId: state.userReducer.user.defaultShippingAddressId,
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		actions: bindActionCreators({
			getCreditCards,
			updateCustomer,
			getCustomerAddresses,
			deleteCustomerAddress,
			deleteCreditCard,
			showAlert,
		}, dispatch),
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(withScreen(AccountDetailsScreen));
