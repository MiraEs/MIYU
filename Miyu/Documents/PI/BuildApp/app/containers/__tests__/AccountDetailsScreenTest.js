
jest.mock('../../../app/services/httpClient', () => ({}));
jest.mock('../../../app/store/configStore', () => ({}));
jest.mock('../../../app/lib/analytics/tracking');
jest.mock('../../actions/NavigatorActions', () => ({
	navigatorPopToRoute: jest.fn(),
}));
jest.mock('../../constants/RouteIdConstants', () => ({
	ACCOUNT_DETAILS_SCREEN: 'ACCOUNT_DETAILS_SCREEN',
}));

import { AccountDetailsScreen } from '../AccountDetailsScreen';
import React from 'react';
import { Alert } from 'react-native';
import { create } from 'react-test-renderer';
import { navigatorPopToRoute } from '../../actions/NavigatorActions';
import { ACCOUNT_DETAILS_SCREEN } from '../../constants/RouteIdConstants';

const addresses = [{
	addressId: 1,
}, {
	addressId: 2,
}];

const fullProps = {
	customer: {},
	creditCards: [],
	actions: {
		trackState: jest.fn(),
		getCreditCards: jest.fn(() => ({ catch: jest.fn(() => ({ done: jest.fn()})) })),
		getCustomerAddresses: jest.fn(() => ({ catch: jest.fn(() => ({ done: jest.fn()})) })),
	},
	navigator: {
		push: jest.fn(),
	},
	addresses,
};

function setup(props = {}) {
	const wrapper = create(
		<AccountDetailsScreen
			{...fullProps}
			{...props}
		/>
	);
	const instance = wrapper.getInstance();
	return {
		wrapper,
		instance,
	};
}

describe('AccountDetailsScreen component', () => {
	it('should render correctly', () => {
		const { wrapper } = setup();
		expect(wrapper.toJSON()).toMatchSnapshot();
	});
	it('should handle componentWillReceiveProps', () => {
		const { instance } = setup();
		instance.componentWillReceiveProps(fullProps);
		expect(instance.state).toMatchSnapshot();
	});
	it('should track the screen', () => {
		const { wrapper } = setup();
		expect(wrapper.getInstance().setScreenTrackingInformation()).toEqual({
			name: 'build:app:more:acct',
		});
	});
	it('should getScreenData', () => {
		const { instance } = setup();
		instance.getScreenData();
		expect(fullProps.actions.getCreditCards).toBeCalled();
		expect(fullProps.actions.getCustomerAddresses).toBeCalled();
	});
	it('should getCreditCards', () => {
		const { instance } = setup();
		expect(instance.getCreditCards(fullProps)).toEqual(fullProps.creditCards);
	});
	it('should getSortedAddresses', () => {
		const { instance } = setup();
		const props = {
			addresses,
		};
		expect(instance.getSortedAddresses(props)).toMatchSnapshot();
	});
	it('should addOrEditAddress', () => {
		const { instance } = setup();
		instance.addOrEditAddress(addresses[0]);
		expect(fullProps.navigator.push).toBeCalledWith('newAddress', {
			address: addresses[0],
			addressScreenState: 'EDIT_ADDRESS',
			title: 'Edit Address',
			addressTypeId: 'SHIPPING',
			onSaveSuccess: instance.navigateToAccountDetails,
		});
	});
	it('should changePassword', () => {
		const { instance } = setup();
		instance.changePassword();
		expect(fullProps.navigator.push).toBeCalledWith('changePassword');
	});
	it('should deleteCreditCard', () => {
		const { instance } = setup();
		const alertSpy = jest.spyOn(Alert, 'alert');
		instance.deleteCreditCard(1);
		expect(alertSpy).toBeCalled();
	});
	it('should deleteCustomerAddress', () => {
		const { instance } = setup();
		const alertSpy = jest.spyOn(Alert, 'alert');
		instance.deleteCustomerAddress(1);
		expect(alertSpy).toBeCalled();
	});
	it('should editDetails', () => {
		const { instance } = setup();
		instance.editDetails();
		expect(fullProps.navigator.push).toBeCalledWith('editAccountDetails');
	});
	it('should onPressNewCardButton', () => {
		const { instance } = setup();
		instance.onPressNewCardButton();
		expect(fullProps.navigator.push).toBeCalledWith('creditCardScreen', {
			hideTabs: false,
			onSaveSuccess: instance.navigateToAccountDetails,
			isCheckout: false,
		});
	});
	it('should navigateToAccountDetails', () => {
		const { instance } = setup();
		instance.navigateToAccountDetails();
		expect(navigatorPopToRoute).toBeCalledWith(ACCOUNT_DETAILS_SCREEN, 'more');
	});
});
