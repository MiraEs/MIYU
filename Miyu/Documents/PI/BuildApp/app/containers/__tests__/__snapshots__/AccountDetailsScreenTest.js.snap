// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDetailsScreen component should getSortedAddresses 1`] = `
Array [
  Object {
    "addressId": 2,
  },
  Object {
    "addressId": 1,
  },
]
`;

exports[`AccountDetailsScreen component should handle componentWillReceiveProps 1`] = `
Object {
  "dataSource": Object {
    "_data": Object {
      "account settings": Array [
        Object {
          "component": [Function],
        },
      ],
      "addresses": Array [
        Object {
          "addressId": 2,
        },
        Object {
          "addressId": 1,
        },
        Object {
          "component": [Function],
        },
      ],
      "email subscriptions": Array [
        Object {
          "component": [Function],
        },
      ],
      "payments": Array [
        Object {
          "component": [Function],
        },
      ],
    },
    "getRowCount": [Function],
  },
  "optimistic": false,
}
`;

exports[`AccountDetailsScreen component should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <ListView
    dataSource={
      Object {
        "_data": Object {
          "account settings": Array [
            Object {
              "component": [Function],
            },
          ],
          "addresses": Array [
            Object {
              "component": [Function],
            },
          ],
          "email subscriptions": Array [
            Object {
              "component": [Function],
            },
          ],
          "payments": Array [
            Object {
              "component": [Function],
            },
          ],
        },
        "getRowCount": [Function],
      }
    }
    enableEmptySections={false}
    renderHeader={[Function]}
    renderRow={[Function]}
    renderSectionHeader={[Function]}
    scrollsToTop={true}
  />
</View>
`;
